### PR TITLE
WIP: Major refactor

### PR DESCRIPTION
**Pull request by mielverkerken**

### CODE REVIEW

File: rustiflow/src/flows/cic_flow.rs [hunk 1]
```diff
@@ -533,13 +528,13 @@ impl CicFlow {
...
     /// * `threshold` - The threshold in microseconds to determine state transitions between active and idle.
-    fn update_active_idle_time(&mut self, timestamp: &Instant, threshold: f64) {
-        if timestamp.duration_since(self.end_active).as_micros() as f64 > threshold {
-            let duration = self.end_active.duration_since(self.start_active);
-            if duration.as_secs_f64() > 0.0 {
-                self.update_active_flow(duration.as_micros() as f64);
+    fn update_active_idle_time(&mut self, timestamp: &DateTime<Utc>, threshold: f64) {
+        if timestamp.signed_duration_since(self.end_active).num_microseconds().unwrap() as f64 > threshold {
+            let duration = self.end_active.signed_duration_since(self.start_active);
+            if !duration.is_zero() {
```
Comment (matissecallewaert) [inline rustiflow/src/flows/cic_flow.rs:534]: Don't know if this would ever occur, but what if the duration is smaller then zero?

**Review by matissecallewaert — APPROVED** [review]
Looks good already! The only thing is testing this I assume?

File: rustiflow/src/flows/cic_flow.rs (hunk 1 above)
Comment (mielverkerken) [reply rustiflow/src/flows/cic_flow.rs:550]: signed_duration_since: Subtracts another DateTime from the current date and time. This does not overflow or underflow at all.

I expect no problems

**Review by matissecallewaert — APPROVED** [review]
Lets rock and roll :rocket:

**Pull request merged**